### PR TITLE
Bug fix: Bypass HTTP exception when adding non existing zones to powerdns

### PR DESCRIPTION
**Pull request by aashifb**
This is in reference to issue https://github.com/github/octodns/issues/437

### CODE REVIEW

File: octodns/provider/powerdns.py [hunk 1]
```diff
@@ -178,7 +178,8 @@ def populate(self, zone, target=False, lenient=False):
                 # Nicer error message for auth problems
                 raise Exception('PowerDNS unauthorized host={}'
                                 .format(self.host))
-            elif e.response.status_code == 422:
+            elif e.response.status_code == 422 or \
+                    e.response.status_code == 404:
```
Comment (ross) [inline octodns/provider/powerdns.py]: Slightly cleaner to do an `in` here. 
```
     elif e.response.status_code in (404, 422):
```

**Comment by ross** [comment]
Actually looks like the tests are failing. That'll need addressed first.

**Comment by aashifb** [comment]
> Actually looks like the tests are failing. That'll need addressed first.

**Comment by aashifb** [comment]
Thanks @ross. I'll look at the tests.

**Comment by yzguy** [comment]
This behavior was fixed in #540, closing this as unnecessary.

**Pull request closed without merging**